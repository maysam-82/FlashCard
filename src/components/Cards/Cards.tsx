import React from 'react';
import Card from '../Card';
import { ICard } from '../../types/card';

export interface IProps {
	cards: ICard[];
}
const Cards: React.SFC<IProps> = ({ cards }) => {
	return (
		<div>
			{cards.map(({ prompt, id, answer }) => (
				<Card key={id} prompt={prompt} answer={answer} />
			))}
		</div>
	);
};

export default Cards;
