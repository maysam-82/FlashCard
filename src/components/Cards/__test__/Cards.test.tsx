import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Cards, { IProps } from '../Cards';

const props: IProps = {
	cards: [
		{
			id: 560,
			prompt: '560 prompt test',
			answer: '560 answer test',
		},
		{
			id: 670,
			prompt: '670 prompt test',
			answer: '670 answer test',
		},
	],
};

describe('Cards component', () => {
	let component: ShallowWrapper;
	beforeEach(() => {
		component = shallow(<Cards {...props} />);
	});
	it('should render Cards', () => {
		expect(component).toMatchSnapshot();
	});
	it('should have two `Card` component', () => {
		expect(component.find('Card').length).toEqual(2);
	});
});
