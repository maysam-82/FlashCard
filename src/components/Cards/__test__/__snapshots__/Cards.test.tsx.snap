// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards component should render Cards 1`] = `
<div>
  <Card
    answer="560 answer test"
    key="560"
    prompt="560 prompt test"
  />
  <Card
    answer="670 answer test"
    key="670"
    prompt="670 prompt test"
  />
</div>
`;
